http-client.ts

```diff
@@ -40,17 +42,23 @@ export class HttpClient implements Server.IHttpClient {
 		let pipeTo = options.pipeTo;
 		delete options.pipeTo;
 
-		let proto = this.$config.USE_PROXY ? "http" : requestProto;
+		const proxyCache = this.$proxyService.getCache();
+		let proto = proxyCache && proxyCache.USE_PROXY ? "http" : requestProto;
 		let http = require(proto);
 
 		options.headers = options.headers || {};
 		let headers = options.headers;
 
-		if (proxySettings || this.$config.USE_PROXY) {
+		if (proxySettings || (proxyCache && proxyCache.USE_PROXY)) {
```

Comment:
We can extract this to it's own method? So that we can make unit tests for it and validate that we don't have some regressions?